### PR TITLE
updating preparing doc + index

### PR DESCRIPTION
**Pull request by choldgraf**
This makes some improvements to the preparing doc so that it's easier to find the information people may be looking for regarding configuration files, and to make more connections with the r2d docs. This touches on #41 though is a bit more general. It also splits up the index a bit so that it's easier to find various sections. The PR adds a full configuration file section (which it pulls from repo2docker to avoid redundancy)

it shouldn't be merged until https://github.com/jupyter/repo2docker/pull/165  and https://github.com/jupyter/repo2docker/pull/168 are merged

built docs here: http://predictablynoisy.com/binder/

### CODE REVIEW

File: doc/parse_readmes.py [hunk 1]
```diff
@@ -18,14 +18,14 @@
...
-repos = org.get_repos()
+repos = list(org.get_repos())
 
 lines = ['# Sample Binder Repositories\n',
          '\n'
          'Below we list several sample Binder repositories that\n'
          'demonstrate how to compose build files in order to create\n'
          'Binders with varying environments.\n\n']
-for repo in repos:
+for ii, repo in enumerate(repos):
```
Comment (willingc) [inline doc/parse_readmes.py]: Let's define what `ii` is or likely even better pick a more meaningful name or abbreviation

File: doc/sample_repos.md [hunk 2]
```diff
@@ -108,8 +118,11 @@ makes use of `pandas`.
 index.ipynb
 requirements.txt
 ```
+```eval_rst
+|
 
-
+```
+---------
 ## Using latex with Binder
```
Comment (willingc) [inline doc/sample_repos.md]: latex -> spelling

File: doc/sample_repos.md [hunk 3]
```diff
@@ -191,26 +210,45 @@ RStudio support with JupyterHub*
 ### Files
 ```
 Dockerfile
+How to Launch RStudio.ipynb
```
Comment (willingc) [inline doc/sample_repos.md]: does this file name have slashes for the spaces?

File: doc/sample_repos.md [hunk 4]
```diff
@@ -191,26 +210,45 @@ RStudio support with JupyterHub*
...
+If you specify `python-2.7` in `runtime.txt`, then:
+
+* A python3 environment is created & installed (this is what the notebook runs from)
+* A python2 environment is created and registered
+* The contents of `requirements.txt` are installed into the python2 environment
+
+Note:
+* Make sure that you save your notebooks with a python 2 kernel activated,
+  as this defines which kernel Binder will use when a notebook is opened.
+* If you *also* wish to install python 3 dependencies, you may do so by including a
```
Comment (willingc) [inline doc/sample_repos.md]: Let's make the first note "important" (must do to have it work) and the second note a "note" (optional)

File: doc/sample_repos.md [hunk 5]
```diff
@@ -238,25 +276,102 @@ index.ipynb
 postBuild
 requirements.txt
 ```
+```eval_rst
+|
+
+```
+---------
+## Using anaconda with pip in the same build
```
Comment (willingc) [inline doc/sample_repos.md]: conda?

File: doc/sample_repos.md [hunk 6]
```diff
@@ -238,25 +276,102 @@ index.ipynb
...
+```eval_rst
+|
 
+```
+---------
+## Creating interactive presentations on Binder with RISE
+
+[![Binder](http://mybinder.org/badge.svg)](http://mybinder.org/v2/gh/binder-examples/jupyter-rise/master?filepath=index.ipynb) | [repo link](https://github.com/binder-examples/jupyter-rise)
 
+In a recent release of RISE, there is support for automatically launching
```
Comment (willingc) [inline doc/sample_repos.md]: let's link to the RISE docs or repo here.

File: doc/sample_repos.md (hunk 2 above)
Comment (choldgraf) [reply doc/sample_repos.md]: You mean LaTeX ? If so, done!

File: doc/sample_repos.md (hunk 3 above)
Comment (choldgraf) [reply doc/sample_repos.md]: ah no - I think instead I'm just going to rename this to `instructions.ipynb`

**Comment by choldgraf** [comment]
ok - I believe comments addressed!

**Comment by betatim** [comment]
Merged!

We should think about the order in the http://predictablynoisy.com/binder/using.html section. Reading it I felt like we oscillate between examples, demos, specs. I have no concrete suggestion right now though...

**Comment by choldgraf** [comment]
Thanks for the comments @willingc!

And I definitely agree @betatim , if you wanna do a few doc iterations w/ me on this repo I am all for it. The Binder docs get about 10x the traffic than either binderhub or repo2docker docs, so I think some more polishing and reorganization would go a long way.

**Pull request merged**